transports/webrtc/src/transport.rs

```diff
@@ -0,0 +1,874 @@
+    type Item = TransportEvent<<WebRTCTransport as Transport>::ListenerUpgrade, Error>;
+
+    fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context) -> Poll<Option<Self::Item>> {
+        if let Some(closed) = self.report_closed.as_mut() {
+            // Listener was closed.
+            // Report the transport event if there is one. On the next iteration, return
+            // `Poll::Ready(None)` to terminate the stream.
+            return Poll::Ready(closed.take());
+        }
+        if let Some(event) = self.poll_if_addr(cx) {
+            return Poll::Ready(Some(event));
+        }
+
+        let udp_mux = self.udp_mux.clone();
+        let udp_mux_read_fut = self
+            .udp_mux_read_fut
+            .get_or_insert(Box::pin(async move { udp_mux.read_from_conn().await }));
+        match udp_mux_read_fut.as_mut().poll(cx) {
+            Poll::Pending => Poll::Pending,
+            Poll::Ready(event) => {
+                self.udp_mux_read_fut = None;
+
+                match event {
+                    UDPMuxEvent::NewAddr(new_addr) => {
+                        let local_addr = socketaddr_to_multiaddr(&self.listen_addr);
+                        let send_back_addr = socketaddr_to_multiaddr(&new_addr.addr);
+                        let event = TransportEvent::Incoming {
+                            upgrade: Box::pin(upgrade::webrtc(
+                                self.udp_mux.clone(),
+                                self.config.clone(),
+                                new_addr.addr,
+                                new_addr.ufrag,
+                                self.id_keys.clone(),
+                            )) as BoxFuture<'static, _>,
+                            local_addr,
+                            send_back_addr,
+                            listener_id: self.listener_id,
+                        };
+                        Poll::Ready(Some(event))
+                    }
+                    UDPMuxEvent::Error(e) => {
+                        self.close(Err(Error::UDPMuxError(e)));
+                        return self.poll_next(cx);
+                    }
+                    _ => return self.poll_next(cx),
+                }
+            }
+        }
+    }
+}
+
+// TODO: remove
+fn hex_to_cow<'a>(s: &str) -> Cow<'a, [u8; 32]> {
+    let mut buf = [0; 32];
+    hex::decode_to_slice(s, &mut buf).unwrap();
+    Cow::Owned(buf)
+}
+
+/// Turns an IP address and port into the corresponding WebRTC multiaddr.
+pub(crate) fn socketaddr_to_multiaddr(socket_addr: &SocketAddr) -> Multiaddr {
+    // TODO: remove
+    let f = "ACD1E533EC271FCDE0275947F4D62A2B2331FF10C9DDE0298EB7B399B4BFF60B";
+    Multiaddr::empty()
+        .with(socket_addr.ip().into())
+        .with(Protocol::Udp(socket_addr.port()))
+        .with(Protocol::XWebRTC(hex_to_cow(&f)))
+}
+
+/// Renders a [`TinyTemplate`] description using the provided arguments.
+pub(crate) fn render_description(
+    description: &str,
```

Comment:
This function is never called with anything else other than a constant, meaning we don't need this parameter at all and can just refer to the constant inside the function body.

Reply:
> This function is never called with anything else other than a constant, meaning we don't need this parameter at all and can just refer to the constant inside the function body.

nope. it's called with a **different** const in upgrade.rs